toolkit/tools/pkg/imagecustomizerlib/runscripts.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safemount"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"golang.org/x/sys/unix"
+)
+
+func runUserScripts(baseConfigPath string, scripts []imagecustomizerapi.Script, listName string,
+	imageChroot *safechroot.Chroot,
+) error {
+	if len(scripts) <= 0 {
+		return nil
+	}
+
+	logger.Log.Infof("Running %s scripts", listName)
+
+	configDirMountPath := filepath.Join(imageChroot.RootDir(), configDirMountPathInChroot)
+
+	// Bind mount the config directory so that the scripts can access any required resources.
+	mount, err := safemount.NewMount(baseConfigPath, configDirMountPath, "", unix.MS_BIND|unix.MS_RDONLY, "", true)
+	if err != nil {
+		return err
+	}
+	defer mount.Close()
+
+	// Runs scripts.
+	for i, script := range scripts {
+		err := runUserScript(i, script, listName, imageChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = mount.CleanClose()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func runUserScript(scriptIndex int, script imagecustomizerapi.Script, listName string,
+	imageChroot *safechroot.Chroot,
+) error {
+	var err error
+
+	scriptLogName := createScriptLogName(scriptIndex, script, listName)
+
+	logger.Log.Infof("Running script (%s)", scriptLogName)
+
+	// Collect the process name and args.
+	tempScriptFullPath := ""
+	process := ""
+	args := []string(nil)
+	if script.Path != "" {
+		scriptPath := filepath.Join(configDirMountPathInChroot, script.Path)
+
+		if script.Interpreter != "" {
+			process = script.Interpreter
+
+			args = []string{scriptPath}
+			args = append(args, script.Arguments...)
+		} else {
+			process = scriptPath
+
+			args = script.Arguments
+		}
+	} else {
+		process = script.Interpreter
+		if process == "" {
```

Comment:
nit: would it make sense to assume the same default in `validateScript` when a script path is specified and no interpreter or execute bits are present?

Reply:
Ok, I've changed it so that `/bin/sh` is now always the default.